realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -57,12 +65,118 @@ void RealSenseNodeFactory::getDevice(rs2::device_list list)
 		else
 		{
 			bool found = false;
+
+			if (_accel_orientation.empty()) // generate accel orientation vector from camera quaternion
+			{
+				try {
+					geometry_msgs::TransformStamped transformStamped = 
+									_tf_buffer.lookupTransform(_tf_camera_link_name, _tf_reference_link_name, ros::Time(0));
+
+					tf2::Quaternion q_gravity_base, q_camera_rot, q_gravity_camera;
+					// universal gravity vector in refence frame of robot in quaternion form
+					q_gravity_base.setValue(0.0, 0.0, 9.81);
+					// fetch pure rotation quaternion Q of sensor relative to robot reference frame
+					tf2::convert(transformStamped.transform.rotation , q_camera_rot);
+					// Calculate the new orientation of gravity in sensor base reference frame
+					// apply quaternion rotation to gravity vector q' = Q * q * Q^-1
+					// https://gamedev.stackexchange.com/questions/28395/rotating-vector3-by-a-quaternion
+					q_gravity_camera = q_camera_rot * q_gravity_base * q_camera_rot.inverse();
+					// transform to accel optical frame reference (uses different convention than robot reference base and camera base)
+					q_gravity_camera.setValue(-q_gravity_camera.y(), -q_gravity_camera.z(), q_gravity_camera.x());
```

Comment:
It actually says [here](https://github.com/IntelRealSense/librealsense/blob/master/doc/d435i.md) that the IMU frame is aligned with the depth (optical) frame, so you ideally you would lookup the transform from `base_link` to `camera_depth_optical_frame` to not have to do this. (You might also consider to add a 4th timeout argument to the lookupTransform call)

Reply:
"camera_depth_optical_frame" is also unavilable unless the sensor is initialized. Using the "base_frame_id" is the only available solution in the sensor initialization code.

[ERROR 1575618742.529330843 ros.realsense2_camera]: "tf_realsense_bottom_depth_optical_frame" passed to lookupTransform argument target_frame does not exist. 
[ERROR 1575618742.529602993 ros.realsense2_camera]: "tf_realsense_top_depth_optical_frame" passed to lookupTransform argument target_frame does not exist.

Reply:
Any suggestions for the timeout duration ? How about 2 secs ?

Reply:
Hmm it seems like that urdf file isn't actually used, apart from some test launch files?

Ok i just checked the code and there is literally a hard-coded `quaternion_optical.setRPY(-M_PI / 2, 0.0, -M_PI / 2);` statement in `calcAndPublishStaticTransform()` ... So I guess we can lower the bar a bit on not hard-coding things.